website/scripts/ros/ros_traffic_signal.js

```diff
@@ -30,119 +30,239 @@ function TrafficSignalInfoList(){
         messageType: M_J2735_SPAT
     });
 
+    listener_sim_clock = new ROSLIB.Topic({
+        ros: g_ros,
+        name: T_SIM_CLOCK,
+        messageType: M_ROS_CLOCK
+    });
+
+    // Variables to track simulation vs real clock
+    let useSimClock = false;
+    let simClockTime = null;
+    let lastRealTime = Date.now();
+    let lastSimTime = null;
+
+    // Subscribe to simulation clock
+    listener_sim_clock.subscribe(function(message) {
+        if (!IsROSBridgeConnected()) {
+            return;
+        }
+
+        // If we receive sim clock data, use it
+        if (message && message.clock) {
+            // nsec is somehow not undefined, but we don't need it
+            const nsec = message.clock.nsec !== undefined ? Number(message.clock.nsec) : 0;
+            useSimClock = true;
+            // Convert ROS time (seconds and nanoseconds) to milliseconds
+            let newSimTime = message.clock.sec * 1000 + Math.floor(nsec) / 1000000;
+            simClockTime = newSimTime;
+            lastSimTime = newSimTime;
+            lastRealTime = Date.now();
+        } else {
+            console.log("[WARN] Sim clock message received but invalid format");
+        }
+    });
+
     let signalStateTracking = 0; //0 -> unavailable
-    let latest_start_time =  Date.now();
+    let latest_start_time = Date.now();
+    let sim_start_time = null;
```

Comment:
What is the difference between sim_start_time and lastSimTime?

Reply:
Please add comments on real and sim time variables.

Reply:
I renamed them so they are clear. I did have unnecessary duplicate code, I apologize.